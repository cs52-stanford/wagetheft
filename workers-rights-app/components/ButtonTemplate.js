import React from "react";
import { TouchableOpacity, Text, StyleSheet } from "react-native";
import PropTypes from "prop-types";
import Colors from "../constants/Colors";

/*
 * Functional Component Defintion: ButtonTemplate
 *
 * This component outlines a basic rounded button for
 * reuse throughout the app.
 *
 */

const ButtonTemplate = ({ style, onPress, titleStyle, title }) => {
  return (
    <TouchableOpacity
      style={{ ...styles.container, ...style }}
      onPress={onPress}
      activeOpacity={0.6}
    >
      <Text style={{ ...styles.titleStyle, ...titleStyle }}>{title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    width: "60%",

    backgroundColor: Colors.darkOrange,
    borderRadius: 8,
    borderStyle: "solid",
    borderWidth: 2,
    borderColor: Colors.darkOrange,
    shadowOpacity: 0.25,
    shadowColor: "black",
    shadowOffset: { width: 0, height: 2 },
    shadowRadius: 6,
    elevation: 3,
    padding: 10,

    position: "absolute",
    bottom: 40,
    justifyContent: "center",
    alignItems: "center",
  },
  titleStyle: {
    color: "white",
  },
});

ButtonTemplate.propTypes = {
  style: PropTypes.object, //eslint-disable-line
  onPress: PropTypes.func.isRequired,
  titleStyle: PropTypes.object, //eslint-disable-line
  title: PropTypes.string.isRequired,
};

export default ButtonTemplate;