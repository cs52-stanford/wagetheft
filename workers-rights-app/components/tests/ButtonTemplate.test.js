import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import ButtonTemplate from "../ButtonTemplate";

describe("<ButtonTemplate/> Tests", () => {
  it("should match snapshot", () => {
    const mockPressFunction = jest.fn();
    const rendered = render(
      <ButtonTemplate title="Test Title" onPress={mockPressFunction} />
    ).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it("should call onPress", () => {
    const mockPressFunction = jest.fn();
    const rendered = render(
      <ButtonTemplate title="Test Title" onPress={mockPressFunction} />
    );

    const buttonTemp = rendered.getByTestId("buttonTemp");
    fireEvent(buttonTemp, "press");
    expect(mockPressFunction).toHaveBeenCalled();
  });
});
