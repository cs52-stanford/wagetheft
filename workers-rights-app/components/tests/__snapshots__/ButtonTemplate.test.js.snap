// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonTemplate/> Tests should match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FC8E56",
      "borderColor": "#FC8E56",
      "borderRadius": 8,
      "borderStyle": "solid",
      "borderWidth": 2,
      "bottom": 40,
      "elevation": 3,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 10,
      "position": "absolute",
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 6,
      "width": "60%",
    }
  }
  testID="buttonTemp"
>
  <Text
    style={
      Object {
        "color": "white",
      }
    }
  >
    Test Title
  </Text>
</View>
`;
